grails-app/services/org/pih/warehouse/report/ReportService.groovy

```diff
@@ -688,6 +688,31 @@ class ReportService implements ApplicationContextAware {
         return data
     }
 
+    List getOnOrderData(String locationId, String productId) {
```

Comment:
Why this was moved out from join to separate query?

Reply:
It is separated because of wrong data returned. On order data should be returned for a product and in previous version it was multiplied.

Reply:
Ok got it, but I am afraid that this solution will result in very poor performance. I'll make some further change requests.